File: Examples/Node.js/Document/CreatePdfFromJPEG.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var PdfApi = require('asposepdfcloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../Data/config.json');
var data_path = '../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Pdf API SDK
var pdfApi = new PdfApi(config);

// Set input file name
var name = "Sample-jpeg.pdf";
var templateFile = "Einstein_JPEG.jpg";
var templateType = "jpeg";

try {
// Upload file to aspose cloud storage
storageApi.PutCreate(templateFile, null, null, data_path + templateFile , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.Pdf Cloud SDK API to create PDF file from JPEG
	pdfApi.PutCreateDocument(name, templateFile, null, templateType, null, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');

			// Download pdf from cloud storage
			storageApi.GetDownload(name, null, null, function(responseMessage) {
				assert.equal(responseMessage.status, 'OK');
				var writeStream = fs.createWriteStream(data_path + "Sample-jpeg_out.pdf");
				writeStream.write(responseMessage.body);
				});
		});
	});

}
catch (e) 
{
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Ruby/Document/create_pdf_from_jpeg.rb
require 'aspose_pdf_cloud'

class Document

  include AsposePDFCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @pdf_api = PdfApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def create_pdf_from_jpeg
    file_name = "sample-jpeg.pdf"
    template_file = "Einstein_JPEG.jpg"
    upload_file(template_file)
    template_type = "jpeg"
    response = @pdf_api.put_create_document(file_name, {template_file: template_file, template_type: template_type})
  end

end

document = Document.new()
puts document.create_pdf_from_jpeg